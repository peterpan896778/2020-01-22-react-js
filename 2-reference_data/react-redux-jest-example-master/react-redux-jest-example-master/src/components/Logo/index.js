import React from 'react';
import styled from 'styled-components';

const StyledLogo = styled.img`
  animation: App-logo-spin infinite 20s linear;
  height: 80px;

  @keyframes App-logo-spin {
    from { transform: rotate(0deg); }
    to { transform: rotate(360deg); }
  };
`;

const Logo = ({ ...props }) => <StyledLogo {...props} />;
export default Logo;
