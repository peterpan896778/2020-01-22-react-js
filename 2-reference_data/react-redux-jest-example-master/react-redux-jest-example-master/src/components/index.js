export { default as Button } from './Button';
export { default as Counter } from './Counter';
export { default as Logo } from './Logo';
