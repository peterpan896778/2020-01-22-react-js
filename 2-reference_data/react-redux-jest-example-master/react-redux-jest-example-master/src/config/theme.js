const colors = {
  blue: '#1565C0',
  blueDark: '#0D47A1',
  green: '#2E7D32',
  greenDark: '#1B5E20',
  red: '#D32F2F',
  redDark: '#B71C1C',
};

const theme = {
  buttons: {
    danger: {
      default: colors.red,
      hover: colors.redDark,
    },
    primary: {
      default: colors.blue,
      hover: colors.blueDark,
    },
    success: {
      default: colors.green,
      hover: colors.greenDark,
    },
  },
  colors,
};

export default theme;
