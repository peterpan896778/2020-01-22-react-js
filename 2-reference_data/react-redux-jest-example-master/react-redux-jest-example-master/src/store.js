import { createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import reducer from 'src/reducers';

const store = createStore(
  reducer,
  composeWithDevTools(),
);

export default store;
