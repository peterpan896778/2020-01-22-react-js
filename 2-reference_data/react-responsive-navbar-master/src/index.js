import ResponsiveNavbar from './responsive-navbar.component';

export default ResponsiveNavbar;
