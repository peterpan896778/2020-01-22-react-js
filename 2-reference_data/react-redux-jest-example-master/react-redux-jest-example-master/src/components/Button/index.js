import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from 'src/config/theme';

/**
 * Button component
 */
const StyledButton = styled.button`
  color: ${props => (
    theme.buttons[props.type].default ||
    theme.buttons.primary.default
  )};
  border: 2px solid ${props => (
    theme.buttons[props.type].default ||
    theme.buttons.primary.default
  )};
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;

  &:hover {
    background-color: ${props => (
      theme.buttons[props.type].hover ||
      theme.buttons.primary.hover
    )};
    color: white;
    border: 2px solid ${props => (
      theme.buttons[props.type].hover ||
      theme.buttons.primary.hover
    )};
  }
`;

const Button = ({ ...props }) => (
  <StyledButton {...props}>{props.label}</StyledButton>
);

Button.defaultProps = {
  type: 'primary',
  label: 'Button',
};

Button.propTypes = {
  /** Button label */
  label: PropTypes.string,
  /** Button type (it can be 'success', 'danger' or 'primary') */
  type: PropTypes.string,
};

export default Button;

