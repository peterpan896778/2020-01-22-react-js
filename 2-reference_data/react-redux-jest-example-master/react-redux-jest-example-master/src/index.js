import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { App } from 'src/containers';

import store from './store';
import registerServiceWorker from './config/registerServiceWorker';

ReactDOM.render(
  <Provider store={store}>
    <App />
  </Provider>,
  document.getElementById('root'),
);

registerServiceWorker();
