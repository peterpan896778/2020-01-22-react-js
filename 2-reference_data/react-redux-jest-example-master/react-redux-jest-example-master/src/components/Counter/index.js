import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'src/components';

import {
  ButtonsContainer,
  CounterContainer,
} from './styles';

const Counter = ({ increment, decrement, value }) => (
  <div>
    <ButtonsContainer>
      <Button
        label="Decrement"
        type="danger"
        onClick={decrement}
      />
      <Button
        label="Increment"
        type="success"
        onClick={increment}
      />
    </ButtonsContainer>
    <CounterContainer>
      <span>Counter: {value}</span>
    </CounterContainer>
  </div>
);

Counter.defaultProps = {
  decrement: () => {},
  increment: () => {},
  value: 0,
};

Counter.propTypes = {
  /** Counter decrement funcion */
  decrement: PropTypes.func,
  /** Counter increment funcion */
  increment: PropTypes.func,
  /** Counter value **/
  value: PropTypes.number,
};

export default Counter;
