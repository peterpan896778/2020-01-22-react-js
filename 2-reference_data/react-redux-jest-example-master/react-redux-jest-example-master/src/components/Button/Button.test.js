import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import theme from 'src/config/theme';
import 'jest-styled-components';

import { Button } from 'src/components';

test('<Button /> (styled-component) mounts without crashing', () => {
  const wrapper = mount(
    <Button />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('<Button /> (styled-component) renders', () => {
  const wrapper = shallow(
    <Button />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('<Button /> (styled-component) renders with type prop', () => {
  const wrapper = shallow(
    <Button type="success" />,
  );
  expect(wrapper).toMatchSnapshot();
});

test('<Button /> (styled-component) renders with type prop and match style rules', () => {
  const wrapper = renderer.create(
    <Button type="success" />,
  ).toJSON();
  expect(wrapper).toMatchSnapshot();
  expect(wrapper).toHaveStyleRule('color', theme.buttons.success.default);
  expect(wrapper).toHaveStyleRule('border', `2px solid ${theme.buttons.success.default}`);
});
