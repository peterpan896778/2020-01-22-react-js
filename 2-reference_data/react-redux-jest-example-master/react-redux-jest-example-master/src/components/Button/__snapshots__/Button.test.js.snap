// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> (styled-component) mounts without crashing 1`] = `
.c0 {
  color: #1565C0;
  border: 2px solid #1565C0;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c0:hover {
  background-color: #0D47A1;
  color: white;
  border: 2px solid #0D47A1;
}

<Button
  label="Button"
  type="primary"
>
  <styled.button
    label="Button"
    type="primary"
  >
    <button
      className="c0"
      label="Button"
      type="primary"
    >
      Button
    </button>
  </styled.button>
</Button>
`;

exports[`<Button /> (styled-component) renders 1`] = `
<styled.button
  label="Button"
  type="primary"
>
  Button
</styled.button>
`;

exports[`<Button /> (styled-component) renders with type prop 1`] = `
<styled.button
  label="Button"
  type="success"
>
  Button
</styled.button>
`;

exports[`<Button /> (styled-component) renders with type prop and match style rules 1`] = `
.c0 {
  color: #2E7D32;
  border: 2px solid #2E7D32;
  background-color: white;
  padding: 10px 20px;
  border-radius: 5px;
  font-size: 1em;
  cursor: pointer;
  outline: 0;
}

.c0:hover {
  background-color: #1B5E20;
  color: white;
  border: 2px solid #1B5E20;
}

<button
  className="c0"
  label="Button"
  type="success"
>
  Button
</button>
`;
